src/legacy/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -55,6 +55,17 @@ export function getUiSettingDefaults() {
     'buildNum': {
       readonly: true
     },
+    'defaultRoute': {
+      name: i18n.translate('kbn.advancedSettings.defaultRoute.defaultRouteTitle', {
+        defaultMessage: 'Default route',
+      }),
+      value: '/app/kibana',
```

Comment:
Since there isn't any validation on the ui settings themselves, it's possible to set the default route to something like `app/canvas` which is relative, but will end up redirecting the user to `http://localhost:5601/spaces/app/canvas`. At a minimum, we should make the necessity to start this with a `/` obvious in the description and "fall-back" to the default in this situation. 

How hard do you think it'd be to add reg-ex based validation to the UI?

Reply:
> At a minimum, we should make the necessity to start this with a / obvious in the description and "fall-back" to the default in this situation.

Good call, added in https://github.com/elastic/kibana/pull/44678/commits/113730901cb30f6edae12d3ffae9e4bb80164779


> How hard do you think it'd be to add reg-ex based validation to the UI?

Hmm, it's an interesting thought. I think it'll get clumsy, as we'd want to specify the regex here in [`ui_setting_defaults`](https://github.com/elastic/kibana/pull/44678/files/95f1ef91960031783dfa33a570d8956c016fa41d#diff-36e5569f6d20405ebbef97de606fb8c8), which is serialized to JSON and sent to the client as part of the set of injected vars.

Serializing a regexp to JSON by default returns `{}`:
```js
JSON.stringify({validation: /asdf/}) // '{"validation": {}}'
```

Calling `toString` beforehand will correctly translate this to a string:
```js
/asdf/.toString() // "/asdf/"
```

Rehydrating the stringified regex is the clumsy part though, as you need to strip off the leading and trailing `/` before sending it into `new RegExp()`.


We _could_ get around all of this by defining the regex as a string without the leading and trailing `/`, so long we we won't need to use any of the regex flags. We can do without them for this specific validation, but I'm not sure how useful this would be outside of this specific field.

Reply:
> We could get around all of this by defining the regex as a string without the leading and trailing /, so long we we won't need to use any of the regex flags. We can do without them for this specific validation, but I'm not sure how useful this would be outside of this specific field.

If we want to go this route, I think that's the cleanest approach. The [RegExp constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) takes separate arguments for the pattern and the flags. It's hard to comment on it's usefulness outside of the current use-case. Ideally, we'd allow a more flexible mechanism of sharing validation between the client and the server. We've gotten away without needing it thus far though. A majority of the other "field types" which are rendered, have at least a minimum about of validation built-in. Booleans are rendered as a check-box, numbers are forced to be numbers, JSON is forced to be properly formatted, etc. 

Since the role of the [default route provider](https://github.com/elastic/kibana/blob/99dab60bf82e7abb6a5b4205b15524a645a44f1a/src/legacy/server/http/setup_default_route_provider.ts) is to enforce our validation on consumption of the setting, I'd prefer we at least add the "begins with a slash" validation here and fall-back to using the default setting if it doesn't.

Reply:
> Since the role of the default route provider is to enforce our validation on consumption of the setting, I'd prefer we at least add the "begins with a slash" validation here and fall-back to using the default setting if it doesn't.

I'll take a closer look and see how the implementation goes. If it turns into its own project, then I'd rather wait for https://github.com/elastic/kibana/issues/46717

Reply:
turns out it wasn't too bad: https://github.com/elastic/kibana/pull/44678/commits/bd43645700169e24d8f1a20916df776790834799

Reply:
<3 reviewing now